tests/cover/test_stateful.py

```diff
@@ -173,15 +173,45 @@ def __init__(self, value):
 class DepthMachine(RuleBasedStateMachine):
     charges = Bundle(u'charges')
 
-    @rule(targets=(charges,), child=charges)
-    @rule(targets=(charges,), child=none())
-    def charge(self, child):
-        return DepthCharge(child)
+    # double-rule is deprecated
+    with Settings(strict=False):
+        @rule(targets=(charges,), child=charges)
+        @rule(targets=(charges,), child=none())
+        def charge(self, child):
+            return DepthCharge(child)
 
     @rule(check=charges)
     def is_not_too_deep(self, check):
         assert check.depth < 3
 
+
+class MultipleRulesSameFuncMachine(RuleBasedStateMachine):
+
+    def myfunc(self, data):
+        print(data)
+
+    rule1 = rule(data=just(u"rule1data"))(myfunc)
+    rule2 = rule(data=just(u"rule2data"))(myfunc)
+
+
+class PreconditionMachine(RuleBasedStateMachine):
+    num = 0
+
+    @rule()
```

Comment:
It's probably not really necessary, but I wonder if it would make sense to also have a rule in this test that resets num to zero. Right now the opportunities for the preconditions firing are a bit thin on the ground.

I don't actually expect this to catch anything but it would increase my confidence in the test as a reader.